receiver/postgresqlreceiver/client.go

```diff
@@ -17,24 +17,23 @@ package postgresqlreceiver // import "github.com/open-telemetry/opentelemetry-co
 import (
 	"context"
 	"database/sql"
-	"errors"
 	"fmt"
 	"net"
 	"strings"
 
 	"github.com/lib/pq"
 	"go.opentelemetry.io/collector/config/confignet"
 	"go.opentelemetry.io/collector/config/configtls"
-	"go.opentelemetry.io/collector/receiver/scrapererror"
+	"go.uber.org/multierr"
 )
 
 type client interface {
 	Close() error
-	getCommitsAndRollbacks(ctx context.Context, databases []string) ([]MetricStat, error)
-	getBackends(ctx context.Context, databases []string) ([]MetricStat, error)
-	getDatabaseSize(ctx context.Context, databases []string) ([]MetricStat, error)
-	getDatabaseTableMetrics(ctx context.Context) ([]MetricStat, error)
-	getBlocksReadByTable(ctx context.Context) ([]MetricStat, error)
+	getDatabaseStats(ctx context.Context, databases []string) (map[string]databaseStats, error)
```

Comment:
For each map that we return, we need to at least have a comment that describes the meaning of the key.

Maybe slightly better is to create a small number of aliases, which make the maps self-documenting and correlate the parameters to the return values.
```go
type databaseName string
type tableName string

getDatabaseStats(ctx context.Context, databases []databaseName) (map[databaseName]databaseStats, error)
getBackends(ctx context.Context, databases []databaseName) (map[databaseName]int64, error)
getDatabaseSize(ctx context.Context, databases []databaseName) (map[databaseName]int64, error)
getDatabaseTableMetrics(ctx context.Context, db databaseName) (map[tableName]tableStats, error)
getBlocksReadByTable(ctx context.Context, db databaseName) (map[tableName]tableIOStats, error)

`